### PR TITLE
Fix protocol-relative URLs in `Url::to()` (#13156)

### LINKED ISSUE #13156: Creating url using yii\helpers\Url::to
There is a issue when you try to create a url passing absolute path and schema in parameters.

For exemple:
`Url::to('@cdn/images/logo.gif', 'http');`
When `@cdn` is an alias for `//cdn.exemple.com`

Will output:
`http://www.example.com/cdn.exemple.com/images/logo.gif`

The lack of some blocks of code after the following commit may be causing this issue:
https://github.com/yiisoft/yii2/commit/2d91723934abac1b9d202eae3827de015b7c927a#diff-477d3f5c7059d9e20a85ef9675f2af3cL221

### PR DESCRIPTION
**Pull request by rob006**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13156

Fix #13156.

### CODE REVIEW

File: framework/helpers/BaseUrl.php [hunk 1]
```diff
@@ -223,7 +223,10 @@ public static function to($url = '', $scheme = false)
             return $url;
         }
 
-        if (($pos = strpos($url, ':')) === false || !ctype_alpha(substr($url, 0, $pos))) {
+        if (
+            substr($url, 0, 2) !== '//'
+            && (($pos = strpos($url, ':')) === false || !ctype_alpha(substr($url, 0, $pos)))
+        ) {
```
Comment (rob006) [inline framework/helpers/BaseUrl.php]: Any idea why we're not using `static::isRelative()` here?

Comment (SilverFire) [inline framework/helpers/BaseUrl.php]: Because `isRelative()` method was added after this line. You can use it

**Review by SilverFire — CHANGES_REQUESTED** [review]

**Review by samdark — CHANGES_REQUESTED** [review]
Please add changelog.

File: framework/helpers/BaseUrl.php (hunk 1 above)
Comment (rob006) [inline framework/helpers/BaseUrl.php]: Done.

**Comment by rob006** [comment]
@samdark This bug is not released, it is related to #13067.

**Review by samdark — APPROVED** [review]

**Review by cebe — APPROVED** [review]

**Review by SilverFire — APPROVED** [review]

**Comment by SilverFire** [comment]
Thank you!~

**Pull request merged**